### PR TITLE
Add a WebAssembly example for ttf-parser

### PR DESCRIPTION
**Pull request by ebraminio**
wished `ttfp_font` was a stack allocated variable which guess isn't possible, hope you find it useful also :)

### CODE REVIEW

**Comment by RazrFalcon** [comment]
Yes, it's looks interesting.

>ttfp_font was a stack allocated variable

Yes, I plan to fix this.

File: c-api/examples/wasm/build.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (RazrFalcon) [inline c-api/examples/wasm/build.sh:1]: I think a readme should be enough.

File: c-api/examples/wasm/index.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        exports.memory.grow(400);
+        // Use heap start as the font address
+        var fontAddress = exports.__heap_base.value + 10000;
+
+        var heapu8 = new Uint8Array(exports.memory.buffer);
+        heapu8.set(fontBlob, fontAddress);
+
+        console.log(exports.ttfp_fonts_in_collection(fontAddress, fontBlob.byteLength));
+    });
+</script>
```
Comment (RazrFalcon) [inline c-api/examples/wasm/index.html:24]: New line.

Comment (ebraminio) [reply c-api/examples/wasm/index.html]: A formatter didn't let me… just removed it, thanks :)

File: c-api/examples/wasm/build.sh (hunk 1 above)
Comment (ebraminio) [reply c-api/examples/wasm/build.sh]: Done.

**Comment by ebraminio** [comment]
> PS: from where the ttc file. What licence does it have?

Oh, is from HarfBuzz of course, 

```
Copyright (c) 2015 by FontTools. No rights reserved.Test TTFRegularFontTools: Test TTF: 2015Version 1.000TestTTF-RegularTest TTF is not a trademark of FontTools.FontToolshttps://github.com/behdad/fonttoolshttps://github.com/behdad/fonttools/blob/master/LICENSE.txt
```

**Comment by ebraminio** [comment]
It is working now and supports fonts drag and drop also, can make it work on nodejs also for automated tests later. Maybe putting it in `./exmaples` instead `./c-api/exmaples` or not? Have a look if it looks good this time, thanks!

File: c-api/examples/wasm/index.html [hunk 3]
```diff
@@ -0,0 +1,60 @@
+<h2>ttf-parser in WebAssembly</h2>
+<p><small>(supports font files drag and drop)</small></p>
+<p><span id="fileName">TTC.ttc</span>:</p>
+<p><code>ttfp_fonts_in_collection():</code> <code id="fontsInCollection"></code></p>
```
Comment (ebraminio) [inline c-api/examples/wasm/index.html:4]: wished `ttfp_fonts_in_collection` could return 1 for non ttc fonts and 0 for non fonts, that sounds more natural to me.

Comment (RazrFalcon) [reply c-api/examples/wasm/index.html:4]: I will think about it. The rust api is more clear since we have `Option`.

**Comment by RazrFalcon** [comment]
I've changed the allocation model, so you can use stack now.

**Comment by ebraminio** [comment]
Just updated it, the new API looks great!

File: c-api/examples/wasm/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+== ttf-parser as WebAssembly module ==
+
+Build:
+
+```
+rustup target add wasm32-unknown-unknown
+(cd ../.. && cargo build --target wasm32-unknown-unknown --release)
```
Comment (RazrFalcon) [inline c-api/examples/wasm/README.md:7]: Could you make 1 command per line? `()` will not work outside the bash.

Comment (RazrFalcon) [reply c-api/examples/wasm/README.md:7]: Or you could use:

```
cargo build --target wasm32-unknown-unknown --release --manifest-path ../../Cargo.toml
```

**Comment by RazrFalcon** [comment]
And I guess we can place the `wasm` dir directly in `ttf-parser/examples`

**Comment by ebraminio** [comment]
All done, feel free to give more feedbacks :)

**Comment by RazrFalcon** [comment]
Everything looks good. Thanks!

**Comment by RazrFalcon** [comment]
How to run it? Should I simple open an `index.html`? Because it errors.

```
index.html:36 Fetch API cannot load file:///home/razr/Projects/rust/ttf-parser/examples/wasm/ttfparser.wasm. URL scheme must be "http" or "https" for CORS request.
(anonymous) @ index.html:36
index.html:36 Uncaught (in promise) TypeError: Failed to fetch
    at index.html:36
```

I'm using the latest Chrome.

**Comment by ebraminio** [comment]
Oh, just yes a local web server, I use one I've built http://pad.js.org/ but guess `python3 -m http.server` should work also, can adopt it with node.js also of course.

**Comment by RazrFalcon** [comment]
Yes, it works fine. Thanks.

**Pull request merged**